extractor/src/main/java/com/example/extractor/ConfigSerializer.java

```diff
@@ -81,64 +79,44 @@ public static String serializeAllInOneConfigToString(final String[] exampleFileP
             final TestInputConfiguration testInputConfiguration = InlineConfigParser.parseWithXmlHeader(exampleFilePath);
             final Configuration xmlConfig = testInputConfiguration.getXmlConfiguration();
             final Configuration targetModule = getTargetModule(xmlConfig);
+
             if (targetModule != null) {
                 isTreeWalker &= isTreeWalkerConfig(xmlConfig);
                 for (final Configuration child : targetModule.getChildren()) {
-                    final Configuration modifiedChild = addIdProperty(child, "example" + (i + 1));
-                    combinedChildren.add(modifiedChild);
+                    final MultiPropertyConfiguration newChild = copyConfiguration(child);
+                    setIdProperty(newChild, "example" + (i + 1));
+                    combinedChildren.add(newChild);
                 }
             }
         }
 
         final String baseIndent = isTreeWalker ? TREE_WALKER_INDENT : NON_TREE_WALKER_INDENT;
         final String combinedModuleContent = buildCombinedModuleChildren(combinedChildren, baseIndent);
-
         final String template = Files.readString(Path.of(templateFilePath), StandardCharsets.UTF_8);
+
         return TemplateProcessor.replacePlaceholders(template, combinedModuleContent, isTreeWalker);
     }
 
-    private static Configuration addIdProperty(final Configuration config, final String idValue) {
-        return new Configuration() {
-            @Override
-            public String getName() {
-                return config.getName();
-            }
-
-            @Override
-            public Map<String, String> getMessages() {
-                return Collections.emptyMap();
-            }
-
-            @Override
-            public String[] getAttributeNames() {
-                return new String[0];
-            }
-
-            @Override
-            public String getAttribute(final String name) {
-                return null;
-            }
-
-            @Override
-            public String[] getPropertyNames() {
-                final List<String> propertyNames = new ArrayList<>(Arrays.asList(config.getPropertyNames()));
-                propertyNames.add("id");
-                return propertyNames.toArray(new String[0]);
-            }
-
-            @Override
-            public String getProperty(final String name) throws CheckstyleException {
-                if ("id".equals(name)) {
-                    return idValue;
+    private static MultiPropertyConfiguration copyConfiguration(final Configuration config) throws CheckstyleException {
+        final MultiPropertyConfiguration newConfig = new MultiPropertyConfiguration(config.getName());
```

Comment:
use DefaultConfiguration class for newConfig.

add onemore parameter to this method `copyConfiguration(final Configuration config, String new Id) ` and set new Id during copy all properties.

Reply:
> use DefaultConfiguration class for newConfig. MultiPropertyConfiguration.java should not be required at all.
> 
> add onemore parameter to this method `copyConfiguration(final Configuration config, String new Id) ` and set new Id during copy all properties.

done